src/controls/shareDialog/ShareDialog.tsx

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import * as React from 'react';
+import {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import { css } from '@emotion/css';
+
+import { RenderDialog } from './renderDialog';
+
+const useStyles = (iFrameHeight: number) => {
+  const styles = {
+    iframe: css({
+      width: "100%",
+      height: iFrameHeight,
+      border: "none",
+    }),
+  };
+  return { styles };
+};
+
+interface IEventData {
+  name: string;
+  command: string;
+  width: number;
+  height: number;
+}
+
+interface IDialogOptions {
+  siteUrl?: string;
```

Comment:
Please review optional properties as looking at the code below it seems that at least some of them are required